Cython/Compiler/Parsing.py

```diff
@@ -3077,18 +3087,49 @@ def p_with_gil(s):
     else:
         return 0
 
-def p_exception_value_clause(s):
+def p_exception_value_clause(s, ctx):
+    """
+    Parse exception value clause.
+
+    Maps clauses to exc_check / exc_value as follows:
```

Comment:
Nit: I would have perhaps put `exc_clause` last, as it's the least important (it only serves to correctly serve the warning for ordering `nogil` incorrectly w.r.t except and friends.

```suggestion
    Maps clauses to exc_check / exc_value / exc_clause as follows:
```

(plus reshuffle table & return value)